### PR TITLE
Fix build with Musl libc

### PR DESCRIPTION
**Pull request by PureTryOut**
The glib header files are available under glib-2.0 but weren't included
as such. This isn't Musl specific but it broke the build

RTLD_DEEPBIND is glibc specific and not in POSIX, guard it's use

TEMP_FAILURE_RETRY is glibc specific. If it's not available, define it.
The implementation is taken from Bionic which is permissively licensed,
so should be compatible with the licensing here but IANAL

### CODE REVIEW

File: multitouch.h [hunk 1]
```diff
@@ -30,6 +30,14 @@
 # include <linux/input.h>
 # include <stdbool.h>
 
+#ifndef input_event_sec
+#define input_event_sec time.tv_sec
```
Comment (FlorentRevest) [inline multitouch.h:34]: Why not define these two macros in musl-compatibility and include that header from multitouch.h and event-input.h ?

**Review by FlorentRevest — APPROVED** [review]

File: multitouch.h (hunk 1 above)
Comment (PureTryOut) [reply multitouch.h:34]: There was a short discussion on this on the upstream SailfishOS side as well. https://git.sailfishos.org/mer-core/mce/merge_requests/148#note_83843

Note that this define isn't actually needed for Musl but only for older kernels which aren't patched with time64 yet on 32-bit arches. `musl-compatibility.h` would be the wrong place, as this on itself doesn't really have anything to do with Musl.

**Comment by PureTryOut** [comment]
Uh, ignore this. I'm upstreaming my fixes to SFOS and will carry this patch downstream for now. Eventually it should trickle down to AsteroidOS sources as well.

**Comment by PureTryOut** [comment]
This has now been merged upstream at https://github.com/sailfishos/mce/pull/1.

**Pull request closed without merging**